test/e2e/serial/tests/workload_unschedulable.go

```diff
@@ -853,31 +852,39 @@ var _ = Describe("[serial][disruptive][scheduler] numaresources workload unsched
 			failedPodIds := e2efixture.WaitForPaddingPodsRunning(fxt, paddingPods)
 			Expect(failedPodIds).To(BeEmpty(), "some padding pods have failed to run")
 
-			//save initial NRT to compare the data after trying to schedule the workloads
-			var err error
-			nrtListInitial, err = e2enrt.GetUpdated(fxt.Client, nrtList, time.Minute)
+			// TODO: interval proportional to periodic update
+			nrtListInitial, err = wait.With(fxt.Client).Interval(11*time.Second).Timeout(1*time.Minute).ForNodeResourceTopologiesSettled(context.TODO(), 3)
 			Expect(err).ToNot(HaveOccurred())
 		})
 
 		AfterEach(func() {
-			By("Verifying NRTs had no updates because the pods failed to be scheduled on any node")
-			nrtListCurrent, err := e2enrt.GetUpdated(fxt.Client, nrtListInitial, time.Second*10)
+			// TODO: interval proportional to periodic update
+			_, err = wait.With(fxt.Client).Interval(11*time.Second).Timeout(1*time.Minute).ForNodeResourceTopologiesSettled(context.TODO(), 3)
 			Expect(err).ToNot(HaveOccurred())
 
-			for _, initialNrt := range nrtListInitial.Items {
-				nrtCurrent, err := e2enrt.FindFromList(nrtListCurrent.Items, initialNrt.Name)
-				Expect(err).ToNot(HaveOccurred())
-
-				dataBefore, err := yaml.Marshal(initialNrt)
-				Expect(err).ToNot(HaveOccurred())
-				dataAfter, err := yaml.Marshal(nrtCurrent)
-				Expect(err).ToNot(HaveOccurred())
-
-				// NRTs before and after should be equal ASSUMING the pods failed scheduling, if not there would be probably a failure in the test steps before this fails
-				ok, err := e2enrt.CheckEqualAvailableResources(initialNrt, *nrtCurrent)
-				Expect(err).ToNot(HaveOccurred())
-				Expect(ok).To(BeTrue(), "NRT of node %q was updated although the pods failed to be scheduled, expected: %s\n  found: %s", initialNrt.Name, dataBefore, dataAfter)
-			}
+			var wg sync.WaitGroup
+			By("Verifying NRTs had no updates because the pods failed to be scheduled on any node")
+			for idx := range nrtListInitial.Items {
+				nrtInitial := &nrtListInitial.Items[idx]
+				wg.Add(1)
+
+				go func(ref *nrtv1alpha2.NodeResourceTopology) {
+					defer GinkgoRecover()
+					defer wg.Done()
+
+					Consistently(func(g Gomega) {
+						var updatedNrt nrtv1alpha2.NodeResourceTopology
+						err := fxt.Client.Get(context.TODO(), client.ObjectKeyFromObject(ref), &updatedNrt)
+						g.Expect(err).ToNot(HaveOccurred(), "cannot get updated NRT for %q", ref.Name)
```

Comment:
This means we'll be fitching the each node NRT in different times. how about that we get the list of the updated NRTs before entering the for loop, and when comparing the objects to the initial we fetch the relevant one by the idx? 
fetching the current NRT object within the loop may find unexpected changes in NRT but I cannot think how can this be helpful in troubleshooting, I even believe it'll make it even hard to spot the root cause because we are checking different versions of NRT overall.

Reply:
I understand what you mean, but I feel depending on the NRT list rather than on specific NRT values is deceptive. NRT objects are updated independently and I'm now very skeptical that checking the list is a good and safe simplification (I may be wrong though). I think that at this stage we should acknowledge the system complexity, so check each NRT specifically, and tame it rather than hide it.

Reply:
no problem, we can keep it like that. 

> NRT objects are updated independently

why? do you mean in code or in actual? because in code I see the list is simply a list of NodeResourceTopology so i'd assume that fetching the NRT "items" from the cluster will get us updated NRT single objects.

Reply:
NRT objects are updated independently by RTE/NFDs, in the live cluster. The intent is to iterate over a list just to get the set of NRT object we should watch to learn when we reach the state we need.

Reply:
(clarified offline) @shajmakh thinking about it, you are probably right about the list. We can implement the same check just checking the list and it's also probably simpler codewise.
Not sure we save time performing individual checks, so we should probably use list and move on.

Reply:
Implemented the change. I'm inclined towards re-evaluating (gradually, over time) changes we made switching to `GetUpdatesForNode`. Most if not all of them seem to be nice (albeit sometimes minor) improvements anyway, so I don't think we need any revert anytime soon. Further fixes and tunings are likely needed and expected.